Makefile

```diff
@@ -40,32 +42,33 @@ dev-all: update-pip  ## Install PECOS in editing mode for development with all o
 uninstall:  ## Uninstall PECOS
 	$(PIP) uninstall quantum-pecos
 
-lint:  ## Run all quality checks / linting
-	pre-commit run --all-files
+lint: metadeps  ## Run all quality checks / linting
+	$(VENV_BIN)/pre-commit run --all-files
 
 docs: install  ## Generate documentation
 	$(PIP) install -r ./docs/requirements.txt
-	cd docs && make clean && make html && cd -
+	cd docs && make clean && make html && cd -  # <<< Will run using the base env... change that make to use .venv?
+# TODO: Maybe call sphinx-build directly...
 
 tests: install  ## Run tests
-	pytest tests
+	$(VENV_BIN)/pytest tests
 
 doctest:  ## Run doctests
-	sphinx-build -b doctest ./docs ./docs/_build
+	$(VENV_BIN)/sphinx-build -b doctest ./docs ./docs/_build
 
 doctest2:  ## Run doctests using pytest
-	pytest ./docs --doctest-glob=*.rst # --doctest-module
+	$(VENV_BIN)/pytest ./docs --doctest-glob=*.rst # --doctest-module
 
 clean:  ## Clean up caches and build artifacts
 	rm -rf *.egg-info dist build docs/_build .pytest_cache/ .ruff_cache/
 
-build: clean  ## Build Python package for upload to pypi
-	python -m build --sdist --wheel -n
-
 metadeps:  ## Install packages used to develop/build this package
 	$(PIP) install -U build pip-tools pre-commit wheel
 
 updatereqs: update-pip  ## Autogenerate requirements
 	$(PIP) install -U pip-tools
 	rm requirements.txt
-	pip-compile --extra=tests --no-annotate --no-emit-index-url --output-file=requirements.txt --strip-extras pyproject.toml
+	$(VENV_BIN)/pip-compile --extra=tests --no-annotate --no-emit-index-url --output-file=requirements.txt --strip-extras pyproject.toml
+
+build: clean venv metadeps  ## Build Python package for upload to pypi
+	$(PYTHON) -m build --sdist --wheel -n
```

Comment:
This doesn't work the first time because `$(BASEPYTHON)` still refers to the old (deleted) Python, see:
```
 ❯ make build
rm -rf *.egg-info dist build docs/_build .pytest_cache/ .ruff_cache/
rm -rf .venv/
~/PECOS/.venv/bin/python3 -m venv .venv
make: /Users/kartik.singhal/GH/PECOS/.venv/bin/python3: No such file or directory
make: *** [venv] Error 1
```
It works the second time. I think it may be simpler to just install `metadeps` and not issue `venv` here because we cannot switch to the virtual environment from a Makefile.

As an aside, we may want to remove the following patterns that throw warnings during build from `MANIFEST.in`:
```
warning: no files found matching '*.c' under directory 'pecos'
warning: no files found matching '*.cpp' under directory 'pecos'
warning: no files found matching '*.h' under directory 'pecos'
warning: no files found matching '*.pyx' under directory 'pecos'
warning: no files found matching '*.pxd' under directory 'pecos'
warning: no files found matching '*.so.*' under directory 'pecos'
```

Reply:
Hmmm, that seems confusing as PYTHON := $(VENV_BIN)/python although maybe there are other environment variables are happening... I'll switch it to be more explict.

I am not sure about removing those patterns. Some of those warnings might be due to the removal of extern/. Right now, I would imagine that extern/ would need to be added manually and in that case MANIFEST is needed to make sure those files are kept when creating the package locally. One solution might be to have the MANIFEST just keep everything in simulators/cuquantum/... or if there is a way to silence the warnings...

Reply:
> PYTHON := $(VENV_BIN)/python

It's actually `$(BASEPYTHON) -m venv $(VENV)` from the execution of the `venv` target.